libcxx/include/flat_set

```diff
@@ -0,0 +1,913 @@
+
+  reverse_iterator rbegin() noexcept { return reverse_iterator(end()); }
+  const_reverse_iterator rbegin() const noexcept { return const_reverse_iterator(end()); }
+  reverse_iterator rend() noexcept { return reverse_iterator(begin()); }
+  const_reverse_iterator rend() const noexcept { return const_reverse_iterator(begin()); }
+
+  const_iterator cbegin() const noexcept { return begin(); }
+  const_iterator cend() const noexcept { return end(); }
+  const_reverse_iterator crbegin() const noexcept { return rbegin(); }
+  const_reverse_iterator crend() const noexcept { return rend(); }
+
+  // capacity
+
+  [[nodiscard]] bool empty() const noexcept { return __c_.empty(); }
+  size_type size() const noexcept { return __c_.size(); }
+  size_type max_size() const noexcept { return __c_.max_size(); }
+
+  // [flat.set.modifiers]
+
+  template <class... _Args>
+  pair<iterator, bool> emplace(_Args&&... __args) {
+    auto __key = key_type(std::forward<_Args>(__args)...);
+    auto __it = lower_bound(__key);
+    if (__it == end() || __compare_(__key, *__it)) {
+      auto __guard = std::__make_exception_guard([&] { __restore_invariant(); });
+      __it = __c_.emplace(__it, std::move(__key));
+      __guard.__complete();
+      return {__it, true};
+    } else {
+      return {__it, false};
+    }
+  }
+
+  template <class... _Args>
+  iterator emplace_hint(const_iterator, _Args&&... __args) {
+    return emplace(std::forward<_Args>(__args)...).first;
+  }
+
+  pair<iterator, bool> insert(const value_type& __x) { return emplace(__x); }
+  pair<iterator, bool> insert(value_type&& __x) { return emplace(std::move(__x)); }
+
+  template <class _K2>
+    requires __is_transparent<_Compare, _K2>::value &&
+             is_constructible_v<value_type, _K2>
+  pair<iterator, bool> insert(_K2&& __x) {
+    auto __it = lower_bound(__x);
+    if (__it == end() || __compare_(__x, *__it)) {
+      auto __guard = std::__make_exception_guard([&] { __restore_invariant(); });
+      __it = __c_.emplace(__it, std::forward<_K2>(__x));
+      __guard.__complete();
+      return {__it, true};
+    } else {
+      return {__it, false};
+    }
+  }
+
+  iterator insert(const_iterator, const value_type& __x) { return emplace(__x).first; }
+  iterator insert(const_iterator, value_type&& __x) { return emplace(std::move(__x)).first; }
+
+  template <class _K2>
+    requires __is_transparent<_Compare, _K2>::value &&
+             is_constructible_v<value_type, _K2>
+  iterator insert(const_iterator, _K2&& __x) {
+    return insert(std::forward<_K2>(__x)).first;
+  }
+
+  template <class _InIt>
+    requires __is_cpp17_input_iterator<_InIt>::value
+  void insert(_InIt __first, _InIt __last) {
+    auto __guard = std::__make_exception_guard([&] { __restore_invariant(); });
+    auto __it = __c_.insert(__c_.end(), __first, __last);
+    auto __end = __c_.end();
+    if (__it != __end) {
+      std::stable_sort(__it, __end, __comp_ref());
```

Comment:
All the places where you use `stable_sort` need the resolution on the issue we discussed yesterday.